Idex: python/debian/changelog
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ python/debian/changelog	2007-03-06 17:09:22.000000000 -0300
@@ -0,0 +1,204 @@
+python2.5 (2.5.1-1osso1) unstable; urgency=low
+
+  * Upgrade to python2.5.1
+  * Moving BaseHTTPServer, SimpleHTTPServer, SimpleXMLRPCServer, CGIHTTPServer, cgi, cgitb to the main package
+
+ -- Luciano Miguel Wolf <luciano.wolf@indt.org.br>  Wed,  2 May 2007 14:50:34 -0300
+
+python2.5 (2.5.0-1osso12) unstable; urgency=low
+
+  * closes #542 and #711
+
+ -- Luciano Miguel Wolf <luciano.wolf@indt.org.br>  Mon, 25 Apr 2007 15:26:00 -0300
+
+python2.5 (2.5.0-1osso11) unstable; urgency=low
+
+  * gdbm dependency added
+
+ -- Luciano Miguel Wolf <luciano.wolf@indt.org.br>  Mon, 19 Mar 2007 16:51:13 -0300
+
+python2.5 (2.5.0-1osso10) unstable; urgency=low
+
+  * leaves pyexpat module inside python package 
+  * bugfix on race condition with signals
+
+ -- Luciano Miguel Wolf <luciano.wolf@indt.org.br>  Thu, 22 Feb 2007 09:15:00 -0300
+
+python2.5 (2.5.0-1osso9) unstable; urgency=low
+
+  * compiled with sqlite3 support
+  * closes #270
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Fri, 23 Jan 2007 14:55:00 -0300
+
+python2.5 (2.5.0-1osso8) unstable; urgency=low
+
+  * adapting the webbrowser module to use the new osso module API
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Fri, 12 Jan 2007 10:34:00 -0300
+
+python2.5 (2.5.0-1osso8) unstable; urgency=low
+
+  * fixed the "MaemoWebBroser" typo in webbrowser module.
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Wed, 10 Jan 2007 17:29:00 -0300
+
+python2.5 (2.5.0-1osso7) unstable; urgency=low
+
+  * put the traceback module back to the runtime package. This module is
+    needed by webbrowser module.
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Fri, 5 Jan 2007 14:35:00 -0300
+
+python2.5 (2.5.0-1osso6) unstable; urgency=low
+
+  * changed "all" to "any" in -dev package (readline.so and other binary
+    modules are platform dependent)
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 27 Dec 2006 10:59:00 -0300
+
+python2.5 (2.5.0-1osso5) unstable; urgency=low
+
+  * added "Replaces: python2.4" to the debian/control file
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 26 Dec 2006 11:12:50 -0300
+
+python2.5 (2.5.0-1osso4) unstable; urgency=low
+
+  * put the readline module in -dev package
+  * added provides python-termios
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Wed, 13 Dec 2006 10:16:11 -0300
+
+python2.5 (2.5.0-1osso3) unstable; urgency=low
+
+  * moving the subprocess module to runtime package
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Wed,  6 Dec 2006 15:29:09 -0300
+
+python2.5 (2.5.0-1osso2) unstable; urgency=low
+
+  * removing readline dependency
+  * removing curses dependency
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Mon, 5 Dec 2006 14:52:00 -0300
+
+python2.5 (2.5.0-1osso1) unstable; urgency=low
+
+  * update to 2.5.0
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Mon, 4 Dec 2006 18:19:00 -0300
+
+python2.5 (2.5c1-1osso1) unstable; urgency=low
+
+  * update to 2.5c1
+  * conflicts with python2.4 packages
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Wed, 19 Jul 2006 18:21:33 -0300
+
+python2.4 (2.4.2-1osso13) unstable; urgency=low
+
+  * Passing $OPT with CFLAGS contents to ./configure so it honors the flags
+
+ -- Elvis Pfutzenreuter <elvis.pfutzenreuter@indt.org.br>  Wed, 14 Jun 2006 15:10:00 -0300
+
+python2.4 (2.4.2-1osso12) unstable; urgency=low
+
+  * debian/rules totally refactored
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Fri,  9 Jun 2006 15:21:36 -0300
+
+python2.4 (2.4.2-1osso11) unstable; urgency=low
+  
+  * fixed depends on shlibs
+  * enable "-mthumb" and "-msoft-abi=softfp" flags on gcc
+  * removing "*.exe" from development package
+  * removing "/tmp" from development package
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Mon, 5 Jun 2006 15:36:00 -0300
+
+python2.4 (2.4.2-1osso10) unstable; urgency=low
+  
+  * move python manpage to development package
+  * improve development package description
+  * remove VPATH from Makefile
+  * remove files and modules licensed under GPL
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 19 May 2006 13:36:00 -0300
+
+python2.4 (2.4.2-1osso9) unstable; urgency=low
+  
+  * fixes in development package
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Wed, 10 May 2006 19:15:00 -0300
+
+python2.4 (2.4.2-1osso8) unstable; urgency=low
+  
+  * modify import.c to not create a new compiled ".pyc/.pyo"
+  * remove xml from python package. Python for maemo will use python-xml
+    package instead.
+  * put xmllib module (used to compile libxml2 bindings and dbus-bindings)
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Wed, 10 May 2006 15:20:00 -0300
+
+python2.4 (2.4.2-1osso7) unstable; urgency=low
+  
+  * more package cleanups
+  * bump release to recompile in 2.0rc7
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 05 May 2006 16:08:00 -0300
+
+python2.4 (2.4.2-1osso6) unstable; urgency=low
+  
+  * removing unneeded patches
+  * removing lib-tk directory from sys.path
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 02 May 2006 16:56:00 -0300
+
+python2.4 (2.4.2-1osso5) unstable; urgency=low
+  
+  * add binary strip
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 04 Apr 2006 10:03:00 -0300
+
+python2.4 (2.4.2-1osso4) unstable; urgency=low
+  
+  * added /usr/bin/* symlinks
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Mon, 03 Apr 2006 17:33:00 -0300
+
+python2.4 (2.4.2-1osso3) unstable; urgency=low
+  
+  * now including *.so.* files
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Mon, 27 Mar 2006 19:07:00 -0300
+
+python2.4 (2.4.2-1osso2) unstable; urgency=low
+  
+  * fixed bug that remove .svn/tmp when building the package
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Wed, 01 Mar 2006 15:58:00 -0300
+
+python2.4 (2.4.2-1osso1) unstable; urgency=low
+  
+  * First OSSO release for SDK 1.2
+
+ -- Ruda Moura <ruda.moura@indt.org.br>  Thu, 09 Feb 2006 10:16:59 -0300
+
+python2.4 (2.4.2-1indt2) unstable; urgency=low
+
+  * Recompiled against SDK 1.1rc8
+
+ -- Ruda Moura <ruda.moura@indt.org.br>  Fri, 02 Dec 2005 15:27:56 -0300
+
+python2.4 (2.4.2-1indt1) unstable; urgency=low
+
+  * Python 2.4.2 release for Pymaemo.
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 25 Oct 2005 17:07:00 -0400
+
+python2.4 (2.4.2-1) unstable; urgency=low
+
+  * Python 2.4.2 release.
+
+ -- Matthias Klose <doko@debian.org>  Thu, 29 Sep 2005 01:49:28 +0200
Index: python/debian/compat
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ python/debian/compat	2007-03-06 17:09:22.000000000 -0300
@@ -0,0 +1 @@
+4
Index: python/debian/control
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ python/debian/control	2007-03-06 17:09:22.000000000 -0300
@@ -0,0 +1,28 @@
+Source: python2.5
+Section: python
+Priority: optional
+Maintainer: Luciano Miguel Wolf <luciano.wolf@indt.org.br>
+Build-Depends: debhelper (>= 4.2), autoconf, zlib1g-dev, libssl-dev, libbz2-dev, libsqlite3-dev (>= 3.3.5-0.2osso1), dpatch, libffi4-dev (>= 4.1.0-1osso1)
+Standards-Version: 3.6.2
+
+Package: python2.5
+Architecture: any
+Depends: ${shlibs:Depends}, libsqlite3-0 (>= 3.3.5-0.2osso1), libffi4 (>= 4.1.0-1osso1)
+Replaces: python2.4
+Conflicts: python2.4
+Provides: python-termios
+Description: An interactive high-level object-oriented language
+ Version 2.5 of the high-level, interactive object oriented language,
+ includes an extensive class library with lots of goodies for
+ network programming, system administration, sounds and graphics.
+
+Package: python2.5-dev
+Architecture: any
+Depends: python2.5 (= ${Source-Version})
+Conflicts: python2.4-dev
+Description: Header files and a static library for Python (v2.5)
+ Header files, a static library and development tools for building
+ Python (v2.5) modules, extending the Python interpreter or embedding
+ Python (v2.5) in applications.
+ .
+ Maintainers of Python packages should read README.maintainers.
Index: python/debian/onlysdk
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ python/debian/onlysdk	2007-03-06 17:09:22.000000000 -0300
@@ -0,0 +1,66 @@
+pdb
+pydoc
+distutils
+compile
+encodings/cp
+encodings/mac
+doctest
+unittest
+hotshot
+config
+dbhash
+curses
+posixfile
+DocXMLRPCServer
+getpass
+macpath
+macurl2path
+os2emxpath
+nturl2path
+pty
+pyclbr
+rexec
+rlcompleter
+robotparser
+smtpd
+statvfs
+stringold
+symtable
+tabnanny
+telnetlib
+this
+timeit
+toaiff
+trace\.py
+tty
+user
+readline
+xdrlib
+LICENSE\.txt
+aifc
+audiodev
+sunaudio
+sunau
+_hotshot.so
+_codecs_tw.so
+_codecs_kr.so
+_codecs_cn.so
+_codecs_hk.so
+_codecs_jp.so
+big5
+big5hkscs
+cp932
+cp949
+cp950
+euc_jis_2004
+euc_jisx0213
+euc_jp
+euc_kr
+gb18030
+gb2312
+gbk
+hz
+johab
+shift_jis
+shift_jis_2004
+shift_jisx0213
Index: python/debian/rules
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ python/debian/rules	2007-03-06 17:09:22.000000000 -0300
@@ -0,0 +1,218 @@
+#!/usr/bin/make -f
+
+PYMAJOR=2
+PYMINOR=5
+PYVER=$(PYMAJOR).$(PYMINOR)
+PYTHON=python$(PYVER)
+
+p_base=$(PYTHON)
+p_dev=$(PYTHON)-dev
+d_base=$(CURDIR)/debian/$(p_base)
+d_dev=$(CURDIR)/debian/$(p_dev)
+
+include /usr/share/quilt/quilt.make
+
+only_dev_list=$(CURDIR)/debian/onlysdk
+
+scriptdir=usr/lib/$(PYTHON)
+
+ifeq ($(DEB_BUILD_ARCH),armel)
+	CFLAGS=-Os -mthumb -mfloat-abi=softfp -Wall -Wstrict-prototypes
+else
+	CFLAGS=-Os -Wall -Wstrict-prototypes
+endif
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f stamp-*
+	-$(MAKE) clean
+	-rm -rf config.log Lib/test/db_home build locales
+	-find -name '*.py[co]' | xargs -n 50 rm -f
+	dh_clean
+
+
+common_configure_args = \
+		--prefix=/usr \
+		--with-fpectl \
+		--enable-ipv6 \
+		--enable-unicode=ucs4 \
+		--without-cxx \
+		--enable-shared \
+		--with-system-ffi \
+		--without-doc-strings \
+		--srcdir=..
+
+stamp-configure:
+	autoconf
+	mkdir -p build
+	cd build && \
+	  CFLAGS="$(CFLAGS)" \
+	  OPT="$(CFLAGS) -DNDEBUG" \
+	  ../configure $(common_configure_args)
+	touch stamp-configure
+
+build: stamp-build
+stamp-build: stamp-configure
+	dh_testdir
+
+	OPT="$(CFLAGS) -DNDEBUG" $(MAKE) -C build
+	touch stamp-build
+
+install: build stamp-install
+stamp-install: stamp-build
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs
+
+	$(MAKE) -C build install DESTDIR=$(d_dev)
+
+	# remove unneeded modules
+	rm -rf \
+	    $(d_dev)/$(scriptdir)/test \
+	    $(d_dev)/$(scriptdir)/ctypes/test \
+	    $(d_dev)/$(scriptdir)/sqlite3/test \
+	    $(d_dev)/$(scriptdir)/email/test \
+	    $(d_dev)/$(scriptdir)/lib-dynload/rgbimage.so \
+	    $(d_dev)/$(scriptdir)/lib-dynload/rgbimg.so \
+	    $(d_dev)/$(scriptdir)/lib-dynload/linuxaudiodev.so \
+	    $(d_dev)/$(scriptdir)/lib-dynload/ossaudiodev.so \
+	    $(d_dev)/$(scriptdir)/lib-dynload/audioop.so \
+	    $(d_dev)/$(scriptdir)/lib-dynload/_ctypes_test.so \
+	    $(d_dev)/$(scriptdir)/lib-dynload/_testcapi.so \
+	    $(d_dev)/$(scriptdir)/idlelib \
+	    $(d_dev)/$(scriptdir)/lib-tk \
+	    $(d_dev)/$(scriptdir)/bsddb \
+	    $(d_dev)/$(scriptdir)/xml/{dom,parsers,sax}
+
+#	$(d_dev)/$(scriptdir)/lib-dynload/pyexpat.so \
+
+	# base
+	# ====
+
+	# create base package directories
+	mkdir -p $(d_base)/$(scriptdir)/plat-linux2
+	mkdir -p $(d_base)/usr/bin
+
+	# shared library
+	mv $(d_dev)/usr/lib/lib$(PYTHON).so.1.0 $(d_base)/usr/lib
+	ln -sf lib$(PYTHON).so.1.0 $(d_base)/usr/lib/lib$(PYTHON).so.1
+
+	# modules that must be outside .zip
+	mv $(d_dev)/$(scriptdir)/site.pyo          $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/os.pyo            $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/posixpath.pyo     $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/stat.pyo          $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/UserDict.pyo      $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/copy_reg.pyo      $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/types.pyo         $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/glob.pyo          $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/fnmatch.pyo       $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/re.pyo            $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/sre.pyo           $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/sre_compile.pyo   $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/sre_constants.pyo $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/sre_parse.pyo     $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/site-packages     $(d_base)/$(scriptdir)
+	mv $(d_dev)/$(scriptdir)/plat-linux2/*.pyo $(d_base)/$(scriptdir)/plat-linux2
+
+	mkdir $(d_base)/$(scriptdir)/lib-dynload
+	for i in `find $(d_dev)/$(scriptdir)/lib-dynload -type f | grep -v -f $(only_dev_list)`; do \
+	    mv $$i $(d_base)/$(scriptdir)/lib-dynload; \
+	done
+
+	(cd $(d_dev)/$(scriptdir) && \
+		find . -name "*.pyo" -type f | \
+		grep -v -f $(only_dev_list) | \
+		zip -0q $(d_base)/usr/lib/python$(PYMAJOR)$(PYMINOR).zip -@ \
+	)
+	find $(d_dev)/$(scriptdir) -name "*.pyo" -type f | grep -v -f $(only_dev_list) | xargs rm -f
+
+	# move the interpreter
+	mv $(d_dev)/usr/bin/python2.5 $(d_base)/usr/bin/python2.5
+
+	# dev
+	# ===
+
+	# scripts
+	mv $(d_dev)/usr/bin/pydoc $(d_dev)/usr/bin/pydoc$(PYVER)
+
+	# manpage
+	mkdir -p $(d_dev)/usr/share/man/man1
+	mv $(d_dev)/usr/man/man1/python.1 $(d_dev)/usr/share/man/man1/$(PYTHON).1
+	rm -rf $(d_dev)/usr/man
+
+	# shared library
+	ln -sf lib$(PYTHON).so.1 $(d_dev)/usr/lib/lib$(PYTHON).so
+	ln -sf ../../lib$(PYTHON).so $(d_dev)/$(scriptdir)/config
+
+	# remove unused items
+	rm -f $(d_dev)/usr/bin/python \
+	      $(d_dev)/usr/bin/idle
+
+	# install the Makefile inside development package
+	sed -e 's,^CXX *=,CXX=		g++ -pthread,' \
+	    -e 's,^RUNSHARED *=.*,RUNSHARED=,' build/Makefile > $(d_dev)/$(scriptdir)/config/Makefile
+
+	# cleanups
+	# ========
+
+	find $(d_base) $(d_dev) -name CVS | xargs rm -rf
+	find $(d_base) $(d_dev) -name .cvsignore | xargs rm -f
+	find $(d_base) $(d_dev) -name "*.exe" | xargs rm -f
+
+	# default python
+	# ============== 
+
+	ln -sf $(PYTHON).1 $(d_dev)/usr/share/man/man1/python.1
+	(cd $(d_base)/usr/bin && ln -sf python$(PYVER) python)
+	ln -sf /usr/lib/python$(PYVER)/pdb.py $(d_dev)/usr/bin/pdb$(PYVER)
+	ln -sf pdb$(PYVER) $(d_dev)/usr/bin/pdb
+	ln -sf /usr/lib/python$(PYVER)/pydoc.py $(d_dev)/usr/bin/pydoc$(PYVER)
+	ln -sf pydoc$(PYVER) $(d_dev)/usr/bin/pydoc
+	cp -p Tools/i18n/pygettext.py $(d_dev)/usr/bin/pygettext$(PYVER)
+	ln -sf pygettext$(PYVER) $(d_dev)/usr/bin/pygettext
+
+	for i in `find $(d_base) $(d_dev) -type f`; do \
+	  sed '1s,#!.*python[^ ]*\(.*\),#!/usr/bin/env $(PYTHON)\1,' $$i > $$i.temp; \
+	  if cmp --quiet $$i $$i.temp; then \
+	    rm -f $$i.temp; \
+	  else \
+	    mv -f $$i.temp $$i; \
+	    chmod 755 $$i; \
+	  fi; \
+	done
+
+	touch stamp-install
+
+binary: binary-indep binary-arch
+binary-indep: install
+	dh_testdir
+	dh_testroot
+	dh_installdirs
+	dh_installexamples
+	dh_link
+	dh_compress -p$(p_dev) -X.py -X.cls -X.css -X.txt -Xgdbinit
+	dh_fixperms
+	dh_installdeb
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary-arch: install
+	dh_testdir
+	dh_testroot
+	dh_installdirs
+	dh_strip
+	dh_link
+	dh_compress -p$(p_dev) -X.py -X.cls -X.css -X.txt -Xgdbinit
+	dh_fixperms
+	dh_makeshlibs
+	dh_installdeb
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+.PHONY: configure build clean binary-indep binary-arch binary install
