Index: python/Modules/Setup.dist
===================================================================
--- python.orig/Modules/Setup.dist	2007-03-06 17:18:48.000000000 -0300
+++ python/Modules/Setup.dist	2007-03-06 17:19:03.000000000 -0300
@@ -96,7 +96,7 @@
 
 # Path component for the Tkinter-related modules
 # The TKPATH variable is always enabled, to save you the effort.
-TKPATH=:lib-tk
+#TKPATH=:lib-tk
 
 COREPYTHONPATH=$(DESTPATH)$(SITEPATH)$(TESTPATH)$(MACHDEPPATH)$(EXTRAMACHDEPPATH)$(TKPATH)
 PYTHONPATH=$(COREPYTHONPATH)
